transaction.go

```diff
@@ -201,7 +201,16 @@ type PackedTransaction struct {
 
 func (p *PackedTransaction) ID() SHA256Bytes {
 	h := sha256.New()
-	_, _ = h.Write(p.PackedTransaction)
+
+	switch p.Compression {
+	case CompressionZlib:
+		r, _ := zlib.NewReader(bytes.NewBuffer(p.PackedTransaction))
+		d, _ := ioutil.ReadAll(r)
+		_, _ = h.Write(d)
+	case CompressionNone:
+		_, _ = h.Write(p.PackedTransaction)
+	}
```

Comment:
Should we also throw like https://github.com/EOSIO/eos/blob/master/libraries/chain/transaction.cpp#L295 if the compression is not none?

Reply:
Maybe we should, but the old `ID()` method returns without error, in fact we just need to calculate correct transaction id when the data is correct. If the compression field is not `none` or `zlib`, it's an undefined behavior and how to return id is up to you.

Reply:
I discuss that with @abourget, and we agreed that the method should now (and probably should have) return an `(SHA256Bytes, error)` and should return an error if the ID cannot be decoded correctly. Can you do it?